models/demos/tg/llama3_70b/demo/demo.py

```diff
@@ -255,14 +307,21 @@ def run_decode(
         input_tokens = tokens[:, prev_pos:cur_pos]
         logits = model.forward(input_tokens, prev_pos)
 
+        if cur_pos == min_prompt_len:
+            total_time_to_first_token_end = time()
+
         next_logits = logits[:, -1, :]  # batch, vocab of last token
         next_token = sampling_func(next_logits)
 
         tokens, eos_reached, prev_pos = prepare_next_input(
             tokenizer, tokens, input_text_mask, finished_mask, prompt_lens, cur_pos, next_token
         )
         latencies.append(time() - start)
```

Comment:
```
if cur_pos == min_prompt_len:
    total_time_to_first_token = time() - start
```
?

Reply:
Prefill stage is done in one iteration of surrounding for loop. model.forward calls prefill_forward which calls prefill for 32 users and total_time_to_first_token is then averaged with num_users later in the code.